src/api/layer.js

```diff
@@ -117,9 +117,7 @@ export default class Layer {
             this._style.onChange(null);
         }
         this._style = style;
-        style.onChange(() => {
-            this._styleChanged(style);
-        });
+        style.onChange(this._styleChanged);
```

Comment:
This seems like a bug to me, `_styleChanged` receives a style argument

Reply:
Since you modify the `onChange` callback to give back the style the lack of an argument is not a problem.

However, I think this lacks a `bind` to `this`

Reply:
I'm not sure about the need of `this`. We are already binding the scope with the first this (`this`._styleChanged)